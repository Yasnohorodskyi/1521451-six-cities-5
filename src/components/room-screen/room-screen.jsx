import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';

import {calcRating} from '../../helpers/calc-rating';
import {premiumTemplate} from '../../helpers/premium-template';
import {connect} from "react-redux";

import ReviewContainer from "../review/review-container";
import MapContainer from '../map/map-container.jsx';
import {MAX_OTHER_REVIEWS} from '../offer/const';
import OfferContainer from '../../components/offer/offer-container.jsx';
import {offerItem} from '../../shapes/offer-item';
import {currentCityShape} from '../../shapes/current-city';
import {getOffer, setFavorite} from '../../store/actions/offers/offers';
import HeaderContainer from '../header/header-container';


class RoomScreen extends PureComponent {
  constructor(props) {
    super(props);
  }
  componentDidMount() {
    const {currentRoom, getOfferDispatch} = this.props;
    getOfferDispatch(currentRoom);
  }
  render() {
    const {offer, nearby, setFavoriteDispatch} = this.props;

    if (offer === null) {
      return null;
    }

    return (

      <div className="page">

        <HeaderContainer />
        <main className="page__main page__main--property">
          <section className="property">
            <div className="property__gallery-container container">
              <div className="property__gallery">
                {offer.images.map((picture, index) => (
                  <div key={index} className="property__image-wrapper">
                    <img className="property__image" src={picture} />
                  </div>
                ))}
              </div>
            </div>
            <div className="property__container container">
              <div className="property__wrapper">
                {
                  (offer.premium) ? premiumTemplate(`property__mark`) : ``
                }
                <div className="property__name-wrapper">
                  <h1 className="property__name">
                    {offer.title}
                  </h1>
                  <button
                    onClick={() => setFavoriteDispatch(offer.id, offer.isFavorite)}
                    className={`property__bookmark-button ${offer.isFavorite} ${offer.isFavorite ? `active` : ``} button`}
                    type="button"
                  >
                    <svg className={`property__bookmark-icon`} width="31" height="33">
                      <use xlinkHref="#icon-bookmark"></use>
                    </svg>
                    <span className="visually-hidden">To bookmarks</span>
                  </button>
                </div>
                <div className="property__rating rating">
                  <div className="property__stars rating__stars">
                    <span className="rating__stars__value" style={calcRating(offer.rating)}></span>
                    <span className="visually-hidden">Rating</span>
                  </div>
                  <span className="property__rating-value rating__value">{offer.rating}</span>
                </div>
                <ul className="property__features">
                  <li className="property__feature property__feature--entire">
                    {offer.type}
                  </li>
                  <li className="property__feature property__feature--bedrooms">
                    {offer.bedrooms}
                  </li>
                  <li className="property__feature property__feature--adults">
                    {offer.maxAdults}
                  </li>
                </ul>
                <div className="property__price">
                  <b className="property__price-value">â‚¬{offer.price}</b>
                  <span className="property__price-text">&nbsp;night</span>
                </div>
                <div className="property__inside">
                  <h2 className="property__inside-title">What&apos;s inside</h2>
                  <ul className="property__inside-list">
                    {offer.goods.map((good, index) => (
                      <li key={index} className="property__inside-item">
                        {good}
                      </li>
                    ))}
                  </ul>
                </div>
                <div className="property__host">
                  <h2 className="property__host-title">Meet the host</h2>
                  <div className="property__host-user user">
                    <img
                      className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
                      src={`/${offer.host.avatar_url}`} alt={offer.host.name}
                    />
                    <span className="property__user-name">
                      {offer.host.name}
                    </span>{offer.isFavorite}
                  </div>
                  <div className="property__description">
                    <p className="property__text">
                      {offer.description}
                    </p>
                  </div>
                </div>
                <section className="property__reviews reviews">
                  <ReviewContainer currentOffer={offer.id} />
                </section>
              </div>
            </div>
            <section className="property__map map">
              <MapContainer
                currentCity={offer.city}
                currentOffer={offer}
                offers={nearby}
              />
            </section>
          </section>
          <div className="container">
            <section className="near-places places">
              <h2 className="near-places__title">Other places in the neighbourhood</h2>
              <OfferContainer
                max={MAX_OTHER_REVIEWS}
                currentCity={offer.city}
                currentOffer={offer.id}
                offers={nearby}
              />
            </section>
          </div>
        </main>
      </div>
    );
  }
}


RoomScreen.propTypes = {
  currentCity: currentCityShape,
  currentRoom: PropTypes.string,
  offers: PropTypes.arrayOf(
      PropTypes.shape({
        offerItem
      })
  ),
  offer: offerItem,
  authorizationStatus: PropTypes.string,
  nearby: PropTypes.arrayOf(
      PropTypes.shape({
        offerItem
      })
  ),
  getOfferDispatch: PropTypes.func,
  setFavoriteDispatch: PropTypes.func,
};


const mapDispatchToProps = (dispatch) => ({
  getOfferDispatch(id) {
    dispatch(getOffer(id));
  },
  setFavoriteDispatch(id, currentStatus) {
    dispatch(setFavorite(id, currentStatus));
  }
});


const mapStateToProps = (state) => {
  return {
    listCities: state.Offers.listCities,
    offer: state.Offers.offer,
    nearby: state.Offers.nearby,
    offers: state.Offers.data,
    authorizationStatus: state.User.authorizationStatus
  };
};

export {RoomScreen};
export default connect(mapStateToProps, mapDispatchToProps)(RoomScreen);
