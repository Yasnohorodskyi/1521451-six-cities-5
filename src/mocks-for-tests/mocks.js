export const offers = [
  {
    city: {
      name: `Hamburg`,
      location: {
        latitude: 53.550343,
        longitude: 10.000654,
        zoom: 13
      }
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`
    ],
    title: `The house among olive `,
    isFavorite: true,
    isPremium: false,
    rating: 4.3,
    type: `house`,
    bedrooms: 5,
    maxAdults: 6,
    price: 190,
    goods: [
      `Washer`,
      `Breakfast`,
      `Laptop friendly workspace`
    ],
    host: {
      id: 25,
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!`,
    location: {
      latitude: 53.53434,
      longitude: 9.998654,
      zoom: 16
    },
    id: 11
  },
  {
    city: {
      name: `Paris`,
      location: {
        latitude: 48.85661,
        longitude: 2.351499,
        zoom: 13
      }
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`
    ],
    title: `Tile House`,
    isFavorite: false,
    rating: 2.9,
    type: `hotel`,
    bedrooms: 2,
    maxAdults: 5,
    price: 266,
    goods: [
      `Air conditioning`,
      `Baby seat`,
      `Towels`,
      `Washer`,
      `Breakfast`,
      `Fridge`,
      `Laptop friendly workspace`
    ],
    host: {
      id: 25,
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    location: {
      latitude: 48.86561,
      longitude: 2.350499,
      zoom: 16
    },
    id: 87
  },
  {
    city: {
      name: `Brussels`,
      location: {
        latitude: 50.846558,
        longitude: 4.351697,
        zoom: 13
      }
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`
    ],
    title: `The house among olive `,
    isFavorite: true,
    isPremium: false,
    rating: 4.6,
    type: `apartment`,
    bedrooms: 4,
    maxAdults: 7,
    price: 218,
    goods: [
      `Washer`,
      `Laptop friendly workspace`,
      `Breakfast`,
      `Air conditioning`,
      `Baby seat`
    ],
    host: {
      id: 25,
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
    location: {
      latitude: 50.86556,
      longitude: 4.371697,
      zoom: 16
    },
    id: 3
  },
  {
    city: {
      name: `Cologne`,
      location: {
        latitude: 50.938362,
        longitude: 6.959974,
        zoom: 13
      }
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`
    ],
    title: `Nice, cozy, warm big bed apartment`,
    isFavorite: true,
    isPremium: false,
    rating: 2.5,
    type: `house`,
    bedrooms: 2,
    maxAdults: 6,
    price: 485,
    goods: [
      `Breakfast`,
      `Washer`,
      `Baby seat`,
      `Laptop friendly workspace`,
      `Air conditioning`
    ],
    host: {
      id: 25,
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.`,
    location: {
      latitude: 50.945362,
      longitude: 6.935974,
      zoom: 16
    },
    id: 34
  }
];

export const cities = {
  Amsterdam: {
    name: `Amsterdam`,
    location: {
      latitude: 52.37454,
      longitude: 4.897976,
      zoom: 13
    }
  },
  Brussels: {
    name: `Brussels`,
    location: {
      latitude: 50.846558,
      longitude: 4.351697,
      zoom: 13
    }
  },
  Cologne: {
    name: `Cologne`,
    location: {
      latitude: 50.938362,
      longitude: 6.959974,
      zoom: 13
    }
  },
  Dusseldorf: {
    name: `Dusseldorf`,
    location: {
      latitude: 51.225403,
      longitude: 6.776314,
      zoom: 13
    }
  },
  Hamburg: {
    name: `Hamburg`,
    location: {
      latitude: 53.550343,
      longitude: 10.000654,
      zoom: 13
    }
  },
  Paris: {
    name: `Paris`,
    location: {
      latitude: 48.85661,
      longitude: 2.351499,
      zoom: 13,
    }
  }
};


export const reviews = [
  {
    id: 18,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 4,
    comment: `вапвпапвапвавапва вапвпапвапвавапвавапвпапвапвавапвавапвпапвапвавапвавапвпапвапвавапва`,
    date: `2020-11-24T14:44:02.210Z`
  },
  {
    id: 17,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 4,
    comment: `Test1 To submit review please make sure to setratingand describe your stay with at least50 characters.\nTo submit review please make sure to setratingand describe your stay with at least50 characters.\n\n\n`,
    date: `2020-11-22T01:32:42.268Z`
  },
  {
    id: 16,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 2,
    comment: `234324324234234221332323213123123123123234324324234234221332323213123123123123234324324234234221332323213123123123123234324324234234221332323213123123123123234324324234234221332323213123123123123234324324234234221332323213123123123123234324324234234221332323213123123123123234324324234234221332323213123123123123234324324234234221332323213123123123123234324324234234221332323213123123123123234324324234234221332323213123123123123234324324234234221332323213123123123123234324324234234221332323213123123123123`,
    date: `2020-11-22T01:32:15.744Z`
  },
  {
    id: 15,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 3,
    comment: `sadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadass`,
    date: `2020-11-22T01:20:16.707Z`
  },
  {
    id: 14,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 4,
    comment: `sadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsada`,
    date: `2020-11-22T01:18:54.165Z`
  },
  {
    id: 13,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 4,
    comment: `efeffqw efeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqw`,
    date: `2020-11-21T16:15:10.563Z`
  },
  {
    id: 12,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 3,
    comment: `The deluxe room was a quite comfortable one with all the adequate facilities. The only thing that made me feel uncomfortable was the rude behavior of an impolite staff at the reception desk. `,
    date: `2020-11-21T16:09:20.677Z`
  },
  {
    id: 11,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 4,
    comment: `123123121231231212312312123123121231231212312312123123121231231212312312123123121231231212312312123123121231231212312312123123121231231212312312123123121231231212312312123123121231231212312312`,
    date: `2020-11-21T16:08:28.941Z`
  },
  {
    id: 10,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 2,
    comment: `wqeqweqweqweqwewqeqwewqeqweqweqweqwewqeqwewqeqweqweqweqwewqeqwewqeqweqweqweqwewqeqwewqeqweqweqweqwewqeqwewqeqweqweqweqwewqeqwewqeqweqweqweqwewqeqwewqeqweqweqweqwewqeqwewqeqweqweqweqwewqeqwe`,
    date: `2020-11-21T16:07:46.202Z`
  },
  {
    id: 9,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 4,
    comment: `textFieldtextFieldtextFieldtextFieldtextFieldtextFieldtextFieldtextFieldtextFieldtextFieldtextFieldtextFieldtextFieldtextFieldtextFieldtextField`,
    date: `2020-11-21T16:07:00.223Z`
  },
  {
    id: 8,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 4,
    comment: `wqewqewqewqewqewqeqweqweqwewqewqeqweqweqweqweqweqweqweqweqweqweqweqweqweqwewqeqweqweqweqwewqeqweqw`,
    date: `2020-11-21T16:05:44.050Z`
  },
  {
    id: 7,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 5,
    comment: `sdsfsdfdsfsdfdsfsdfdsf`,
    date: `2020-11-21T15:37:35.160Z`
  },
  {
    id: 6,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 5,
    comment: `qweqweqweqweqweqweqdsdsdsdqwdqwdweqweqweqweqwdwqddwdwq`,
    date: `2020-11-21T14:48:36.841Z`
  },
  {
    id: 5,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 5,
    comment: `qweqweqweqweqweqweqwewqeqwedweweqewqewqewqewqeqwewqwq`,
    date: `2020-11-21T14:47:26.390Z`
  },
  {
    id: 4,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 5,
    comment: `fsfsfsd`,
    date: `2020-11-16T17:26:36.123Z`
  },
  {
    id: 3,
    user: {
      id: 15,
      isPro: false,
      name: `Kendall`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/6.jpg`
    },
    rating: 3,
    comment: `I stayed here for one night and it was an unpleasant experience.`,
    date: `2020-11-01T13:38:44.752Z`
  },
  {
    id: 2,
    user: {
      id: 12,
      isPro: true,
      name: `Isaac`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/3.jpg`
    },
    rating: 3,
    comment: `The deluxe room was a quite comfortable one with all the adequate facilities. The only thing that made me feel uncomfortable was the rude behavior of an impolite staff at the reception desk.`,
    date: `2020-11-01T13:38:44.752Z`
  },
  {
    id: 1,
    user: {
      id: 14,
      isPro: true,
      name: `Corey`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/5.jpg`
    },
    rating: 5,
    comment: `The house is very good, very happy, hygienic and simple living conditions around it are also very good. I hope to have the opportunity to come back. Thank you.`,
    date: `2020-11-01T13:38:44.752Z`
  }
];


export const user = {
  authorizationStatus: `AUTH`,
  data: {
    avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`,
    email: `r-sasha@list.ru`,
    id: 1,
    isPro: false,
    name: ``
  },
  error: ``
};


export const titleInputs = [
  `terribly`,
  `badly`,
  `not bad`,
  `good`,
  `perfect`
];


export const optionList = {
  FILTER_POPULAR: `Popular`,
  FILTER_TOP_RATED_FIRST: `Top rated first`,
  FILTER_PRICE_LOW_TO_HIGH: `Price: low to high`,
  FILTER_PRICE_HIGH_TO_LOW: `Price: high to low`
};


export const baseFilter = `Popular`;
