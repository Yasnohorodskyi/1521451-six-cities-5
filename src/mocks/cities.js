export default [
  {
    id: `1`,
    title: `Amsterdam`,
    link: `Amsterdam`,
    lat: 52.3809453943508,
    lon: 4.931309666406198
  },
  {
    id: `2`,
    title: `Hamburg`,
    link: `Hamburg`,
    lat: 53.5611,
    lon: 9.9933
  },
  {
    id: `3`,
    title: `Dusseldorf`,
    link: `Dusseldorf`,
    lat: 51.2277,
    lon: 6.7735
  },
  {
    id: `4`,
    title: `Brussels`,
    link: `Brussels`,
    lat: 50.8503,
    lon: 4.3517
  },
  {
    id: `5`,
    title: `Cologne`,
    link: `Cologne`,
    lat: 50.9375,
    lon: 6.9603
  },
  {
    id: `6`,
    title: `Paris`,
    link: `Paris`,
    lat: 48.8566,
    lon: 2.3522
  },
];
