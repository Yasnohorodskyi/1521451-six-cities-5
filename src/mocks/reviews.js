export default [
  {
    id: `001`,
    offers: 1,
    user: {
      id: `1`,
      name: `Max`,
      ava: `/img/avatar-max.jpg`
    },
    text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
    rating: 2,
    timestamp: 1555793053
  },
  {
    id: `002`,
    offers: 1,
    user: {
      id: `2`,
      name: `Иван`,
      ava: `/img/avatar-max.jpg`
    },
    text: `Отличный вариант для семейных пар!`,
    rating: 4,
    timestamp: 1555793053
  },
];
