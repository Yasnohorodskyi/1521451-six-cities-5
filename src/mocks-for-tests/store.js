import configureStore from "redux-mock-store";
import {offers, cities, reviews, user, baseFilter} from "./mocks";
import {offersAdapter, offerAdapter} from "../helpers/offers-adapter";

const createStore = configureStore();

const mockStore = createStore({
  Offers: {
    currentCity: cities[`Brussels`],
    baseFilter,
    offers: offersAdapter(offers),
    offer: offerAdapter(offers[0]),
    nearby: offersAdapter(offers),
    favorites: offersAdapter(offers),
    data: offersAdapter(offers),
    getFavoriteDispatch: ()=>{}
  },
  Review: {
    reviews
  },
  User: user,
  Offer: offerAdapter(offers[0])
});

export {mockStore};
