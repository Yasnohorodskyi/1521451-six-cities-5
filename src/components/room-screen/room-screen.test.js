import ShallowRenderer from "react-test-renderer/shallow";
import React from "react";

import {RoomScreen} from "./room-screen";
import {offers} from "../../mocks-for-tests/mocks";


it(`RoomScreen`, () => {
  const renderer = new ShallowRenderer();
  const tree = renderer.render(
      <RoomScreen offer={offers[0]} />
  );
  expect(tree).toMatchSnapshot();
});

