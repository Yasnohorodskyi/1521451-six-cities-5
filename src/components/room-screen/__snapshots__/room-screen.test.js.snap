// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomScreen 1`] = `
<div
  className="page"
>
  <Memo(Connect(HeaderContainer)) />
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
          >
            <img
              className="property__image"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              className="property__image"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              className="property__image"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              className="property__image"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              className="property__image"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              className="property__image"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              className="property__image"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              className="property__image"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              className="property__image"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              className="property__image"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              className="property__image"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              className="property__image"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              className="property__image"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              className="property__image"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              The house among olive 
            </h1>
            <button
              className="property__bookmark-button true active button"
              onClick={[Function]}
              type="button"
            >
              <svg
                className="property__bookmark-icon"
                height="33"
                width="31"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="property__rating rating"
          >
            <div
              className="property__stars rating__stars"
            >
              <span
                className="rating__stars__value"
                style={
                  Object {
                    "width": "86%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
            <span
              className="property__rating-value rating__value"
            >
              4.3
            </span>
          </div>
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              house
            </li>
            <li
              className="property__feature property__feature--bedrooms"
            >
              5
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              6
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              190
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <li
                className="property__inside-item"
              >
                Washer
              </li>
              <li
                className="property__inside-item"
              >
                Breakfast
              </li>
              <li
                className="property__inside-item"
              >
                Laptop friendly workspace
              </li>
            </ul>
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <div
                className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
              />
              <span
                className="property__user-name"
              >
                Angelina
              </span>
              true
            </div>
            <div
              className="property__description"
            >
              <p
                className="property__text"
              >
                I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!
              </p>
            </div>
          </div>
          <section
            className="property__reviews reviews"
          >
            <Memo(Connect(ReviewContainer))
              currentOffer={11}
            />
          </section>
        </div>
      </div>
      <section
        className="property__map map"
      >
        <Memo(Connect(MapContainer))
          currentCity={
            Object {
              "location": Object {
                "latitude": 53.550343,
                "longitude": 10.000654,
                "zoom": 13,
              },
              "name": "Hamburg",
            }
          }
          currentOffer={11}
          max={3}
        />
      </section>
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <Memo(Connect(OfferContainer))
          currentCity={
            Object {
              "location": Object {
                "latitude": 53.550343,
                "longitude": 10.000654,
                "zoom": 13,
              },
              "name": "Hamburg",
            }
          }
          currentOffer={11}
          max={3}
        />
      </section>
    </div>
  </main>
</div>
`;
